controller/api/destination/server.go

```diff
@@ -344,56 +349,107 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 // If the pod argument is provided, the profile sent to the client will
 // include an endpoint. Otherwise, the default profile is sent.
 func (s *server) sendEndpointProfile(stream pb.Destination_GetProfileServer, pod *corev1.Pod, port uint32) error {
-	log := s.log
-	var endpoint *pb.WeightedAddr
-	opaquePorts := make(map[uint32]struct{})
-	var err error
-	if pod != nil {
-		podSet := podToAddressSet(s.k8sAPI, pod).WithPort(port)
-		podID := watcher.PodID{
-			Namespace: pod.Namespace,
-			Name:      pod.Name,
-		}
-		var ok bool
-		opaquePorts, ok, err = getPodOpaquePortsAnnotations(pod)
-		if err != nil {
-			log.Errorf("failed to get opaque ports annotation for pod: %s", err)
-		}
+	if pod == nil {
+		translator := newProfileTranslator(stream, s.log, "", port, nil)
+		translator.Update(nil)
+		return nil
+	}
```

Comment:
This isn't new logic. It just allows `sendEndpointProfile` to return early and reduce the nesting below.